sumpy/e2e.py

```diff
@@ -533,55 +534,37 @@ class M2LGenerateTranslationClassesDependentData(E2EBase):
     """
     default_name = "m2l_generate_translation_classes_dependent_data"
 
-    def get_translation_loopy_insns(self, result_dtype):
-        from sumpy.symbolic import make_sym_vector
-        dvec = make_sym_vector("d", self.dim)
-
-        src_rscale = sym.Symbol("src_rscale")
-
-        m2l_translation = self.tgt_expansion.m2l_translation
-
-        from sumpy.assignment_collection import SymbolicAssignmentCollection
-        sac = SymbolicAssignmentCollection()
-        tgt_coeff_names = [
-            sac.assign_unique(
-                f"m2l_translation_classes_dependent_expr{i}", coeff_i)
-            for i, coeff_i in enumerate(
-                m2l_translation.translation_classes_dependent_data(
-                    self.tgt_expansion, self.src_expansion, src_rscale,
-                    dvec, sac=sac))]
-
-        sac.run_global_cse()
-
-        from sumpy.codegen import to_loopy_insns
-        return to_loopy_insns(
-                sac.assignments.items(),
-                vector_names={"d"},
-                pymbolic_expr_maps=[self.tgt_expansion.get_code_transformer()],
-                retain_names=tgt_coeff_names,
-                complex_dtype=to_complex_dtype(result_dtype),
-                )
-
     def get_kernel(self, result_dtype):
+        m2l_translation = self.tgt_expansion.m2l_translation
         m2l_translation_classes_dependent_ndata = \
-            self.tgt_expansion.m2l_translation.translation_classes_dependent_ndata(
+            m2l_translation.translation_classes_dependent_ndata(
                 self.tgt_expansion, self.src_expansion)
+
+        child_knl = \
+            m2l_translation.translation_classes_dependent_data_loopy_knl(
+                self.tgt_expansion, self.src_expansion, result_dtype)
+
         from sumpy.tools import gather_loopy_arguments
         loopy_knl = lp.make_kernel(
                 [
                     "{[itr_class]: 0<=itr_class<ntranslation_classes}",
                     "{[idim]: 0<=idim<dim}",
+                    "{[idata]: 0<=idata<m2l_translation_classes_dependent_ndata}",
                     ],
                 ["""
                 for itr_class
                     <> d[idim] = m2l_translation_vectors[idim, \
-                            itr_class + translation_classes_level_start]
-
-                    """] + self.get_translation_loopy_insns(result_dtype) + ["""
-                    m2l_translation_classes_dependent_data[itr_class, {idx}] = \
-                            m2l_translation_classes_dependent_expr{idx}
-                    """.format(idx=i) for i in range(
-                        m2l_translation_classes_dependent_ndata)] + ["""
+                            itr_class + translation_classes_level_start] \
+                            {id=set_d,dup=idim}
+                    <> coeff[idata] = 0  {id=init,dup=idata}
+                    [idata]: coeff[idata] = \
+                        m2l_data_inner(
```

Comment:
Maybe just `m2l_data`?

Reply:
Fixed in 50f8bb3c643db4c2084cd849db0a9d585f686a42